edgexfoundry/app-functions-sdk-go
# feat: Expose the RequestTimeout configuration setting to app service

Added the RequestTimeout() method on the ApplicationService Interface
with returns a time.Duration. The questTimeout configuration setting
is pared during initialztion so error can be caught during start-up.

Signed-off-by: Leonard Goodell <leonard.goodell@intel.com>







**If your build fails**  due to your commit message not passing the build checks, please review the guidelines here: https://github.com/edgexfoundry/app-functions-sdk-go/blob/main/.github/CONTRIBUTING.md

## PR Checklist
Please check if your PR fulfills the following requirements:

- [ ] I have added unit tests for the new feature or bug fix (if not, why?) **N/A - Initialize can't be unit tested**
      https://github.com/edgexfoundry/edgex-docs/pull/689

## Testing Instructions
Add the following line app template main.go on this branch (around line 65)
```
	app.lc.Infof("Request Timeout is set to: %v", app.service.RequestTimeout())
```
Build and run app template
Verify the following is in the logs:
```
level=INFO ts=2022-02-02T22:18:43.2279695Z app=new-app-service source=main.go:65 msg="Request Timeout is set to: 5s"
```
Modify `Service.RequestTimout` in res/configuration.toml to be set to `"5"`
Rerun App Template and verify it fails to start-up with the following message:
```
level=ERROR ts=2022-02-02T22:29:13.1180933Z app=new-app-service source=factory.go:43 msg="App Service initialization failed: unable to parse Service.RequestTimeout configuration as a time duration: time: missing unit in duration \"5\""
```

## New Dependency Instructions (If applicable)
**N/A**
